src/deprecated.jl

```diff
@@ -2,3 +2,40 @@
 @deprecate get_vel_t velf(state).t
 
 @deprecate propagate(veh, state, roadway, Δt) propagate(veh, state, a, roadway, Δt)
+
```

Comment:
Can you add the `rec` version of `simulate!` in there two?

Reply:
just add a deprecation warning to the `rec` version but leave them

Reply:
Deprecated all versions of `simulate!` using `rec` for which we currently don't have an alternative implementation.